doc/introduction/compiling_circuits.rst

```diff
@@ -397,10 +407,40 @@ JIT compile the full optimization loop, for example when training models:
 
         return params
 
-Finally, Catalyst provides additional features to PennyLane, such as classical
-control of quantum operations that are JIT-enabled, via the function
-``catalyst.for_loop`` and ``catalyst.cond``. It also enables arbitrary
-post-processing of mid-circuit measurements.
+The Catalyst compiler also supports capturing imperative Python control flow
+in compiled programs, resulting in control flow being interpreted at runtime
+rather than in Python at compile time. You can enable this feature via the
+``autograph=True`` keyword argument.
+
+.. code-block:: python
+
+    @qml.qjit(autograph=True)
+    @qml.qnode(dev)
+    def circuit(x: int):
+
+        if x < 5:
+            qml.Hadamard(wires=0)
+        else:
+            qml.T(wires=0)
+
+        return qml.expval(qml.PauliZ(0))
 
-For more details, see the `Catalyst documentation and tutorials
-<https://docs.pennylane.ai/projects/catalyst/>`__.
+>>> circuit(3)
+array(0.)
+>>> circuit(5)
+array(1.)
+
+Note that AutoGraph results in additional restrictions, in particular whenever
+global state is involved.
+Please refer to the :doc:`AutoGraph guide<catalyst:dev/autograph>` for a
+complete discussion of the supported and unsupported use-cases.
+
+For more details on using the :func:`~.qjit` decorator and Catalyst
+with PennyLane, please refer to the Catalyst
+:doc:`quickstart guide <catalyst:dev/quick_start>`, as well as the :doc:`sharp
+bits and debugging tips <catalyst:dev/sharp_bits>` page for an overview of
+the differences between Catalyst and PennyLane, and how to best structure
+your workflows to improve performance when using Catalyst.
+
+To make your own compiler compatible with PennyLane, please see
+the :mod:`~.compiler#adding-a-compiler` module documentation.
```

Comment:
```suggestion
To make your own compiler compatible with PennyLane, please see
the :mod:`~.compiler` module documentation.
```

This won't work unfortunately, and will break the link. I'll come back to this page of the docs once all the compiler PRs are in :)